### PR TITLE
Create synth script for pubsub

### PR DESCRIPTION
**Pull request by dazuma**
Create a synth script for pubsub, and regenerate gapic.

Most changes are docs-related and/or additive. Unit and acceptance tests still pass. Open questions:

* The generated code now provides a google/cloud/pubsub/v1/credentials.rb file that has slightly different scopes from the current hand-written google/cloud/pubsub/credentials.rb. What to do, what to do...
* The newly generated low-level classes doc says the low-level client is alpha, whereas it used to be beta. Do we care?

### CODE REVIEW

File: google-cloud-pubsub/lib/google/cloud/pubsub/v1/doc/google/pubsub/v1/pubsub.rb [hunk 1]
```diff
@@ -477,13 +549,27 @@ class UpdateSnapshotRequest; end
...
-      #     exists -- will expire in 4 days.
+      #     exists -- will expire in 4 days. The service will refuse to create a
+      #     snapshot that would expire in less than 1 hour after creation.
       # @!attribute [rw] labels
       #   @return [Hash{String => String}]
       #     User labels.
       class Snapshot; end
 
-      # Request for the +ListSnapshots+ method.
+      # Request for the GetSnapshot method.<br><br>
```
Comment (blowmage) [inline google-cloud-pubsub/lib/google/cloud/pubsub/v1/doc/google/pubsub/v1/pubsub.rb]: It is strange to see HTML in the documentation. Is it set to output Markdown? I notice that it is using + where I would expect `.

**Comment by blowmage** [comment]
All these changes look good to me. I have one concern about non-Markdown markup, but that might be out of scope for this PR.

File: google-cloud-pubsub/lib/google/cloud/pubsub/v1/doc/google/pubsub/v1/pubsub.rb (hunk 1 above)
Comment (dazuma) [reply google-cloud-pubsub/lib/google/cloud/pubsub/v1/doc/google/pubsub/v1/pubsub.rb]: I noticed that too. The HTML is present in the source proto. I'm actually not sure what the style guidelines are for proto documentation formatting (and I'll try to track that down), but that would be out of scope here.

**Comment by dazuma** [comment]
@blowmage I made one more change here: I made the generated v1/credentials.rb use the same scopes as our current handwritten credentials, and then made our handwritten credentials point back at the generated one. (https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/2237/commits/3ad429c8131d573809da6eda270bbc26bf7a1077). The handwritten credentials now "subclasses" the generated one (rather than just setting the constant) so that documentation will still appear as expected, because I expect that for this library we still want "Pubsub::Credentials" (rather than "Pubsub::V1::Credentials") to be the public API. Does this all sound okay to you?

**Comment by dazuma** [comment]
Aaaaand one more change. Removed the autogenerated "overview.rb" file which contains autogen documentation for `Google::Cloud::Pubsub` that conflicts with our handwritten documentation.

**Comment by blowmage** [comment]
When we deprecated the VideoIntelligence Credentials we aliased the constant instead of using inheritance. Whatever solution we choose should be consistent across gems IMO.

**Comment by blowmage** [comment]
Okay.

**Pull request merged**